src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -552,9 +576,119 @@ private static ITaskSeriesTimer StartParameterLogTimer(IRecurrentCommand updateC
                 await TryHandleTimeoutAsync(invokeTask, CancellationToken.None, throwOnTimeout, timeoutTokenSource.Token, timerInterval, instance, null);
             }
 
-            object returnValue = await invokeTask;
+            if (invokeTask != null)
+            {
+                object returnValue = await invokeTask;
 
-            parameterHelper.SetReturnValue(returnValue);
+                parameterHelper.SetReturnValue(returnValue);
+            }
+        }
+
+        internal static async Task<object> InvokeWithFiltersAsync(List<IFunctionInvocationFilter> filters, IFunctionInvoker invoker, ParameterHelper parameterHelper, object[] invokeParameters,
+            CancellationTokenSource functionCancellationTokenSource, IFunctionInstance instance,
+            ILogger logger, JobHost jobHost, JobHostConfiguration config)
+        {
+            FunctionResult functionResult = null;
+            Exception invokeException = null;
+
+            // Create a dictionary that will pass the user properties from the executing filter to the executed filter
+            Dictionary<string, object> properties = new Dictionary<string, object>();
+
+            Task<object> invokeTask = null;
+            try
+            {
+                await InvokeExecutingFilters(filters, parameterHelper, properties, functionCancellationTokenSource, instance, logger, jobHost, config);
+                invokeTask = invoker.InvokeAsync(invokeParameters);
+                functionResult = new FunctionResult(true);
+            }
+            catch (Exception ex)
+            {
+                invokeException = ex;
+                functionResult = new FunctionResult(ex);
+                throw;
+            }
+            finally
+            {
+                await InvokeExecutedFilters(filters, parameterHelper, properties, functionCancellationTokenSource, instance, logger, jobHost, config, functionResult);
+                if (invokeException != null)
+                {
+                    throw invokeException;
+                }
+            }
+
+            return invokeTask;
```

Comment:
I believe this is incorrect - we will return a Task<Task<object>>

Do we want to wait for all filters to execute, or execute the filters outside of the async context? This question doesn't even make sense to me, so maybe we should chat it out